lib/rex/socket.rb

```diff
@@ -186,26 +186,14 @@ def self.getaddresses(hostname, accept_ipv6 = true)
       return [hostname]
     end
 
-    res = ::Socket.gethostbyname(hostname)
-    return [] if not res
+    res = ::Addrinfo.getaddrinfo(hostname, 0, ::Socket::AF_UNSPEC, ::Socket::SOCK_STREAM)
```

Comment:
Looks like this no longer behaves the same way when it's passed `nil`:
https://github.com/rapid7/rex-sslscan/pull/3#discussion_r712395800